### DISCUSSION TITLE
Excel export of view modified in FetchXMLBuilder

### DISCUSSION THREAD

**Discussion by PeterKCinteros** [rappen/FetchXMLBuilder#974] [General]
Hi
I have modified the conditions of a view with FetchXMLBuilder. The view works nicely within Dynamics. It is not possible to edit the view within Dynamics so I know I have to use FetchXMLBuilder to modify it.

But today I discovered that it is not possible to export this view as a static Excel sheet. 

I have modified views with FetchXMLBuilder for many years but never encountered any problem to export to Excel. Maybe I never before tried to export them to Excel. Are there any known limitations on the possibilities to modify the FetchXml when using the standard Excel export functionality?

/ Peter

**Comment by rappen** [comment]
Hi Peter, and sorry I've been quite busy this fall, and that's why late response...

And "sorry" number 2 😂 - I very rarely use the Excel export - do you have any more info about your view that fails?

**Reply by PeterKCinteros** [reply]
Hi Jonas. 

I have no problem publishing the fetch here.
I have done an or-filter where the conditions are from two different entities. I suspect that this might cause the issue.

`<fetch version="1.0" output-format="xml-platform" mapping="logical" distinct="false">
  <entity name="cint_mms_res_subledger_row">
    <attribute name="cint_name" />
    <order attribute="cint_name" descending="false" />
    <filter type="and">
      <condition attribute="statecode" operator="eq" value="0" />
      <condition attribute="cint_article_no" operator="eq" value="301" />
      <condition attribute="cint_date_accounting" operator="last-month" />
      <filter type="or">
        <condition entityname="fpr" attribute="cint_article_no" operator="not-null" />
        <condition entityname="ppart" attribute="cint_article_no" operator="not-null" />
      </filter>
    </filter>
    <attribute name="cint_payment_status" />
    <attribute name="cint_amount_due" />
    <attribute name="cint_subledger_id" />
    <attribute name="cint_date_accounting" />
    <attribute name="cint_amount" />
    <attribute name="cint_article_no" />
    <attribute name="cint_article_name" />
    <attribute name="cint_mms_res_subledger_rowid" />
    <link-entity name="cint_mms_res_subledger_row" from="cint_mms_res_subledger_rowid" to="cint_full_payment_subledger_row_id" link-type="outer" alias="fpr">
      <attribute name="cint_article_no" />
      <filter>
        <condition attribute="cint_article_no" operator="not-in">
          <value>411</value>
          <value>412</value>
        </condition>
      </filter>
    </link-entity>
    <link-entity name="cint_mms_res_partial_payment" from="cint_payment_subledger_row_id" to="cint_mms_res_subledger_rowid" link-type="outer" alias="pp">
      <link-entity name="cint_mms_res_subledger_row" from="cint_mms_res_subledger_rowid" to="cint_debt_subledger_row_id" link-type="outer" alias="ppart">
        <attribute name="cint_article_no" />
        <filter>
          <condition attribute="cint_article_no" operator="not-begin-with" value="18" />
        </filter>
      </link-entity>
    </link-entity>
  </entity>
</fetch>`

/ Peter

**Reply by rappen** [reply]
Right.
Maybe/probably outer link-entities and filtering on the that's the problem...
Have you searched on Bing and/or [MS Learn](https://learn.microsoft.com/en-us/power-platform/)?

**Reply by PeterKCinteros** [reply]
I have tried searching, and the closest I got was tip 1211 on crmtipoftheday which uses the same kind of query. But it does not mention anything about Excel export unfortunately. But I recreated that tip in a different environment, and then the Excel export worked. The main difference is that the environment where it works is OnPremise version 9.1 and the environment where it does not work in OnLine. I will do some more investigations.
